pkg/tcpip/network/ipv4/ipv4.go

```diff
@@ -200,16 +205,17 @@ func (e *endpoint) writePacketFragments(r *stack.Route, gso *stack.GSO, hdr buff
 
 // WritePacket writes a packet to the given destination address and protocol.
 func (e *endpoint) WritePacket(r *stack.Route, gso *stack.GSO, hdr buffer.Prependable, payload buffer.VectorisedView, params stack.NetworkHeaderParams, loop stack.PacketLooping) *tcpip.Error {
-	ip := header.IPv4(hdr.Prepend(header.IPv4MinimumSize))
+	ip := header.IPv4(hdr.Prepend(header.IPv4MinimumSize + len(options)))
```

Comment:
options is not passed in?

Reply:
if options are not nil, automatically options as to be included in ipv4 header for reply packets, so  we adding length of options also, and it is not going affect any other packets.